### PR TITLE
feat: group workspace logs inside CI logs

### PR DESCRIPTION
**Pull request by nachoaldamav**

### CODE REVIEW

**Review by zkochan — COMMENTED** [review]
maybe we should also print what script is being run?

**Comment by nachoaldamav** [comment]
> maybe we should also print what script is being run?

Yes, I forgot about that, I'm on it

File: exec/plugin-commands-script-runners/src/ciLogs.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    } else if (CI.GITLAB) {
+      return {
+        start: `section_start:${Math.floor(Date.now() / 1000)}:${prefix}\\r\\e[0K${ScriptName}`,
+        end: `section_end:${Math.floor(Date.now() / 1000)}:${prefix}\\r\\e[0K`,
+      }
+    } else if (CI.TRAVIS) {
+      return {
+        start: `travis_fold:start:${prefix}${ScriptName}\r\n`,
+        end: `travis_fold:end:${prefix}${ScriptName}\r\n`,
+      }
```
Comment (zkochan) [inline exec/plugin-commands-script-runners/src/ciLogs.ts]: why is prefix printed twice for gitlab and travis?

Comment (nachoaldamav) [reply exec/plugin-commands-script-runners/src/ciLogs.ts]: True, on it

Comment (nachoaldamav) [reply exec/plugin-commands-script-runners/src/ciLogs.ts]: For gitlab I think the prefix itself is not printed as it's an ID, I'm going to add a property for ID stuff

Comment (zkochan) [reply exec/plugin-commands-script-runners/src/ciLogs.ts]: can't we just duplicate "script name" as id then?

Comment (nachoaldamav) [reply exec/plugin-commands-script-runners/src/ciLogs.ts]: Maybe, but the documentation uses `my_first_section` as the id/name, so idk if the id can use spaces in it, no objections on using the `ScriptName` as the id tho

**Comment by zkochan** [comment]
I would do the next improvements.

The section name calculation should happen outside of the class. In that case, we could reuse the code for other stuff in the future. For instance, logging different parts of installation.

The class would have only one public function: logStart. It would return a logEnd function that would already have all the necessary context for closing the group.

The id of the section can be changed to be `section${counter}`, where the counter is automatically increased by the logger.

**Comment by zkochan** [comment]
@zirkelc it probably makes sense to set reporter-hide-prefix to true, when the output is grouped on CI, right?

**Comment by zirkelc** [comment]
@zkochan yes, I think that would make sense. If the output from a script contains an annotation (e.g. [GitHub Actions annotations](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message), that would get noticed by the CI. 

However, as the grouping is also done with annotations, scripts might be able to mess with the grouping by logging group annotations themselves:
```
::group::{title}
::endgroup::
```

So maybe it should be possible to opt-in or opt-out of grouping on CI?

**Comment by zkochan** [comment]
I don't know. Can groups be nested?

**Comment by zkochan** [comment]
opting out sounds fine by me

**Comment by zirkelc** [comment]
> I don't know. Can groups be nested?

Not yet, but there is a feature request for this: https://github.com/actions/toolkit/issues/1001

**Review by zkochan — APPROVED** [review]

**Pull request merged**